### PR TITLE
Refactor: Email verification

### PR DESCRIPTION
**Pull request by kalleep**
**What is this feature?**
In https://github.com/grafana/grafana/pull/82914 we added some logic to perform email verification on user updates.

We have other cases where we want to reuse this flow so to make that part easier I extracted the verification generation to its own service so we can reuse the logic that before existing only in http handlers.

I first tried to add this function to `user.Service` interface but pulling in `tempuser.Service` and `notifications.Service` created a mess with wrire.

Part of #https://github.com/grafana/identity-access-team/issues/536

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/services/user/userimpl/verifier_test.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+		err := verifier.VerifyEmail(context.Background(), user.VerifyEmailCommand{
+			User:   user.User{ID: 2},
+			Email:  "some@email.com",
+			Action: user.EmailUpdateAction,
+		})
+
+		assert.ErrorIs(t, err, user.ErrUserAlreadyExists)
+	})
+
+	t.Run("should success when no user has the email", func(t *testing.T) {
```
Comment (IevaVasiljeva) [inline pkg/services/user/userimpl/verifier_test.go:39]:
```suggestion
	t.Run("should succeed when no user has the email", func(t *testing.T) {
```

File: pkg/services/user/userimpl/verifier_test.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+			Action: user.EmailUpdateAction,
+		})
+
+		assert.ErrorIs(t, err, nil)
+		assert.True(t, c.expireCalled)
+		assert.True(t, c.createCalled)
+		assert.True(t, c.updateCalled)
+	})
+
+	t.Run("should success user holding email is the same that want to verify it", func(t *testing.T) {
```
Comment (IevaVasiljeva) [inline pkg/services/user/userimpl/verifier_test.go:74]:
```suggestion
	t.Run("should succeed when the user holding the email is the same user that want to verify it", func(t *testing.T) {
```

**Review by IevaVasiljeva — APPROVED** [review]
LGTM

**Review by linoman — COMMENTED** [review]
Great job! I left a minor comment.

You can improve the coverage of the tests by testing for the errors the verifier code might return.

File: pkg/server/wireexts_oss.go [hunk 3]
```diff
@@ -9,10 +9,7 @@ import (
 
 	"github.com/grafana/grafana/pkg/infra/metrics"
 	"github.com/grafana/grafana/pkg/plugins"
-	"github.com/grafana/grafana/pkg/plugins/manager"
-	"github.com/grafana/grafana/pkg/registry"
-	"github.com/grafana/grafana/pkg/registry/backgroundsvcs"
-	"github.com/grafana/grafana/pkg/registry/usagestatssvcs"
+	"github.com/grafana/grafana/pkg/plugins/manager" "github.com/grafana/grafana/pkg/registry" "github.com/grafana/grafana/pkg/registry/backgroundsvcs" "github.com/grafana/grafana/pkg/registry/usagestatssvcs"
```
[the remark this answers is not in the document]
Comment (kalleep) [reply pkg/server/wireexts_oss.go]: Nice, added that one by mistake

**Pull request merged**